### PR TITLE
Initial implementation of NativeInput plugin

### PR DESCRIPTION
**Pull request by ryantrem**
This is an initial implementation of NativeInput. With this change, it only supports touch input (we can add more later, but this puts the basic infrastructure in place). The main layers are:
1. `NativeInput` is the public native contract. It only exposes the ability to *write* input events (e.g. forward them from the platform specific APIs).
2. `NativeInput::Impl` is the internal contract for both *writing* input events, and *reading* (polling) input state.
3. `DeviceInputSystem` is the native "polyfill" (replacement) for the Babylon.js `DeviceInputSystem`. Instances of this NAPI based class simply read input state from the `NativeInput::Impl` instance.

Note that currently, a `NativeInput` instance is scoped to a `Napi::Env`. In the future (as noted in the comments), these should be scoped to individual views that the user can interact with.

See https://github.com/BabylonJS/BabylonReactNative/pull/7 for an example of a native app/integration building on top of `NativeInput`.

FYI @polygonalsun

### CODE REVIEW

File: Plugins/NativeInput/Source/DeviceInputSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#include "DeviceInputSystem.h"
+
+namespace Babylon::Plugins
```
Comment (syntheticmagus) [inline Plugins/NativeInput/Source/DeviceInputSystem.cpp:3]: I think the convention we're establishing is that classes like this should live in the `Babylon::Plugins::Internal` namespace. @bghgary, please correct me if I'm mistaken about that.

https://github.com/BabylonJS/BabylonNative/blob/master/Plugins/NativeWindow/Source/NativeWindow.cpp#L3

File: Plugins/NativeInput/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+add_library(NativeInput ${SOURCES})
+
+target_include_directories(NativeInput PUBLIC "Include")
+
+target_link_to_dependencies(NativeInput
+    PUBLIC JsRuntime
+    PRIVATE arcana)
+
+set_property(TARGET NativeInput PROPERTY FOLDER Polyfills)
```
[the remark this answers is not in the document]
Comment (ryantrem) [reply Plugins/NativeInput/CMakeLists.txt:16]: Yes, it was in Polyfills originally and I moved it to Plugins. Why doesn't this break anything? What is this for?

[the remark this answers is not in the document]
Comment (bghgary) [reply Plugins/NativeInput/CMakeLists.txt:16]: It just defines the filter in VS.

File: Plugins/NativeInput/Include/Babylon/Plugins/NativeInput.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        static NativeInput& CreateForJavaScript(Napi::Env);
+        static NativeInput& GetFromJavaScript(Napi::Env);
+
+        void PointerDown(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y);
+        void PointerUp(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y);
+        void PointerMove(uint32_t pointerId, uint32_t x, uint32_t y);
+
+    protected:
+        NativeInput(const NativeInput&) = delete;
+        NativeInput(NativeInput&&) = delete;
```
[the remark this answers is not in the document]
Comment (ryantrem) [reply Plugins/NativeInput/Include/Babylon/Plugins/NativeInput.h:21]: Oops, left over from a copy and paste. 🙈

File: Plugins/NativeInput/Source/DeviceInputSystem.cpp (hunk 1 above)
Comment (ryantrem) [reply Plugins/NativeInput/Source/DeviceInputSystem.cpp:3]: Well in this case, `DeviceInputSystem` is a private nested class of `Impl` which is a private nested class of `NativeInput`, so I can't change the namespace. These are just implementation details of `NativeInput`.

**Pull request merged**